data_discovery_ai/model/keywordModel.py

```diff
@@ -29,10 +25,13 @@
 from tensorflow.keras.models import load_model
 
 import logging
-from matplotlib import pyplot as plt
-from datetime import datetime
 from typing import Dict, Callable, Any, Tuple, Optional, List
 import os
+from pathlib import Path
+
+# Base directory where your Poetry project's pyproject.toml is located
+BASE_DIR = Path(__file__).resolve().parent.parent
```

Comment:
same, this `BASE_DIR = Path(__file__).resolve().parent.parent` appearing 3 times, need to consolidate this in 1 place.